src/query/xasl_cache.c

```diff
@@ -2125,15 +2191,15 @@ xcache_cleanup (THREAD_ENTRY * thread_p)
       while ((xcache_entry = iter.iterate ()) != NULL)
 	{
 	  candidate.xid = xcache_entry->xasl_id;
-	  candidate.time_last_used = xcache_entry->time_last_used;
-
-	  if (candidate.xid.cache_flag & XCACHE_ENTRY_FLAGS_MASK)
+	  candidate.xcache = xcache_entry;
+	  if (candidate.xid.cache_flag > 0 || (candidate.xid.cache_flag & XCACHE_ENTRY_FLAGS_MASK))
 	    {
-	      /* Either marked for delete or recompile, or already recompiled. Not a valid candidate. */
+	      /* Either marked for delete or recompile, fixed (prepared)
+	         or already recompiled. Not a valid candidate. */
 	      continue;
 	    }
 
-	  (void) bh_try_insert (bh, &candidate, NULL);
+	  bh_try_insert (bh, &candidate, NULL);
```

Comment:
`(void)` is to suppress a warning in build.

Reply:
Ok, thanks